src/vmm/src/resources.rs

```diff
@@ -278,3 +278,296 @@ impl VmResources {
+
+        assert_eq!(actual_boot_cfg, expected_boot_cfg);
+    }
+
+    #[test]
+    fn test_set_boot_source() {
+        let tmp_file = TempFile::new().unwrap();
+        let cmdline = "reboot=k panic=1 pci=off nomodules 8250.nr_uarts=0";
+        let expected_boot_cfg = BootSourceConfig {
+            kernel_image_path: String::from(tmp_file.as_path().to_str().unwrap()),
+            initrd_path: Some(String::from(tmp_file.as_path().to_str().unwrap())),
+            boot_args: Some(cmdline.to_string()),
+        };
+
+        let mut vm_resources = default_vm_resources();
+        let boot_cfg = vm_resources.boot_source().unwrap();
+        let tmp_ino = tmp_file.as_file().metadata().unwrap().st_ino();
+
+        assert_ne!(boot_cfg.cmdline.as_str(), cmdline);
+        assert_ne!(boot_cfg.kernel_file.metadata().unwrap().st_ino(), tmp_ino);
+        assert_ne!(
+            boot_cfg
+                .initrd_file
+                .as_ref()
+                .unwrap()
+                .metadata()
+                .unwrap()
+                .st_ino(),
+            tmp_ino
+        );
+
+        vm_resources.set_boot_source(expected_boot_cfg).unwrap();
+        let boot_cfg = vm_resources.boot_source().unwrap();
+        assert_eq!(boot_cfg.cmdline.as_str(), cmdline);
+        assert_eq!(boot_cfg.kernel_file.metadata().unwrap().st_ino(), tmp_ino);
+        assert_eq!(
+            boot_cfg
+                .initrd_file
+                .as_ref()
+                .unwrap()
+                .metadata()
+                .unwrap()
+                .st_ino(),
+            tmp_ino
+        );
+    }
+
+    #[test]
+    fn test_set_block_device() {
+        let mut vm_resources = default_vm_resources();
+        let mut new_block_device_cfg = vm_resources.block.config_list.get(0).unwrap().clone();
+        let tmp_file = TempFile::new().unwrap();
+        new_block_device_cfg.drive_id = "block3".to_string();
+        new_block_device_cfg.path_on_host = tmp_file.as_path().to_path_buf();
+        assert_eq!(vm_resources.block.config_list.len(), 1);
+        vm_resources.set_block_device(new_block_device_cfg).unwrap();
+        assert_eq!(vm_resources.block.config_list.len(), 2);
+    }
+
+    #[test]
+    fn test_set_vsock_device() {
+        let mut vm_resources = default_vm_resources();
+        let new_vsock_cfg = VsockDeviceConfig {
+            vsock_id: "new_vsock".to_string(),
+            guest_cid: 1,
+            uds_path: String::from(TempFile::new().unwrap().as_path().to_str().unwrap()),
+        };
+        assert!(vm_resources.vsock.is_none());
+        vm_resources.set_vsock_device(new_vsock_cfg.clone());
+        let actual_vsock_cfg = vm_resources.vsock.as_ref().unwrap().clone();
+        assert_eq!(actual_vsock_cfg, new_vsock_cfg);
+    }
+
+    #[test]
+    fn test_set_net_device() {
+        let mut vm_resources = default_vm_resources();
+        let mut new_net_device_cfg = vm_resources
```

Comment:
Why not simply create a `NetworkInterfaceConfig` instance?

Reply:
I prefer to use what it is already there and clone it, instead of repeating the creation of the entire struct.